python/ray/ml/predictor.py

```diff
@@ -27,7 +28,27 @@ def from_checkpoint(cls, checkpoint: Checkpoint) -> "Predictor":
         """
         raise NotImplementedError
 
-    def predict(self, data: DataBatchType) -> DataBatchType:
+    def predict(
+        self, preprocessed_dataset: ray.data.Dataset, **iter_batches_kwargs
+    ) -> "pd.DataFrame":
+
+        """Run inference on dataset.
+
+        Args:
+            preprocessed_dataset (Dataset): A Ray dataset that has been
+                preprocessed already.
+            iter_batches_kwargs: kwargs to pass to ``dataset.iter_batches()``
+
+        Returns:
+            A Pandas DataFrame containing prediction values.
+
+        """
+        predictions = []
+        for batch in preprocessed_dataset.iter_batches(**iter_batches_kwargs):
+            predictions.append(self.score_fn(batch))
+        return pd.DataFrame({"predictions": predictions})
```

Comment:
Should we use `map_batches` here for parallel processing?

Reply:
This is now removed!